app/lib/common/widgets/typing_indicator.dart

```diff
@@ -6,29 +8,38 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'dart:async';
 
 class TypingIndicator extends ConsumerWidget {
-  const TypingIndicator({super.key, required this.typingUsers});
+  const TypingIndicator({super.key, required this.roomId});
 
-  final List<AvatarInfo> typingUsers;
+  final String roomId;
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final theme = Theme.of(context).typingIndicatorTheme;
 
-    return TypingWidget(typingUsers: typingUsers, theme: theme);
+    return TypingWidget(roomId: roomId, theme: theme);
   }
 }
 
 /// Typing Widget.
-class TypingWidget extends StatelessWidget {
-  const TypingWidget({
-    super.key,
-    required this.typingUsers,
-    required this.theme,
-  });
+class TypingWidget extends ConsumerWidget {
+  static const typingRendererKey = Key('typing_widget_renderer');
+  const TypingWidget({super.key, required this.roomId, required this.theme});
 
-  final List<AvatarInfo> typingUsers;
+  final String roomId;
   final TypingIndicatorTheme theme;
 
+  List<AvatarInfo> _getTypingUsers(WidgetRef ref) {
+    final users = ref.watch(chatTypingEventProvider(roomId)).valueOrNull;
+    final userIds = users?.map((userId) => userId.toString()).toList() ?? [];
+    return userIds
+        .map(
+          (userId) => ref.watch(
+            memberAvatarInfoProvider((userId: userId, roomId: roomId)),
+          ),
+        )
+        .toList();
+  }
```

Comment:
why this complexity if that isn't actually used (here)?

Reply:
I've created separate high level providers to inject them directly where they are needed :)